File: Framework/Domain/Dtos/Common/ForgotPasswordModel.py

from pydantic import BaseModel
from typing import Optional

# Translate ForgotPasswordResponse
class ForgotPasswordResponse(BaseModel):
    """
    Pydantic model equivalent to the C# ForgotPasswordResponse DTO.
    """
    email: str

# Translate ForgotPasswordResult
class ForgotPasswordResult(BaseModel):
    """
    Pydantic model equivalent to the C# ForgotPasswordResult DTO.
    """
    message: str

# Translate ValidateTokenPasswordResponse
class ValidateTokenPasswordResponse(BaseModel):
    """
    Pydantic model equivalent to the C# ValidateTokenPasswordResponse DTO.
    """
    token: str

# Translate ValidateTokenPasswordResult
class ValidateTokenPasswordResult(BaseModel):
    """
    Pydantic model equivalent to the C# ValidateTokenPasswordResult DTO.
    """
    email: str

# Translate ChangePasswordResponse (based on the snippet, it seems to have a string property, assuming Message)
class ChangePasswordResponse(BaseModel):
    """
    Pydantic model equivalent to the C# ChangePasswordResponse DTO.
    """
    message: str

File: Framework.Domain/Dtos/Common/ForgotPasswordModel.cs
namespace Framework.Domain.Dtos.Common
{
    public class ForgotPasswordResponse
    {
        public string Email { get; set; }
    }

    public class ForgotPasswordResult
    {
        public string Message { get; set; }
    }

    public class ValidateTokenPasswordResponse
    {
        public string Token { get; set; }
    }

    public class ValidateTokenPasswordResult
    {
        public string Email { get; set; }
    }

    public class ChangePasswordResponse
    {
        public string CurrentPassword { get; set; }
        public string NewPassword { get; set; }
    }

    public class ChangePasswordResult
    {
        public string Message { get; set; }
    }

    public class ChangePasswordByForgotResponse
    {
        public string Guid { get; set; }
        public string NewPassword { get; set; }
    }

    public class ChangePasswordByForgotResult
    {
        public string Message { get; set; }
    }
}
